rosidl_generator_c/resource/idl__struct.h.em

```diff
@@ -55,25 +59,33 @@ TEMPLATE(
 from rosidl_parser.definition import Service
 }@
 @[for service in content.get_elements_of_type(Service)]@
+
+@{ hash_var = idl_structure_type_to_c_typename(service.namespaced_type) + '__TYPE_VERSION_HASH' }@
+#define @(hash_var)__INIT @(type_hash_to_c_definition(type_hash['service'], line_final_backslash=True))
+static const rosidl_type_hash_t @(hash_var) = @(hash_var)__INIT;
```

Comment:
It is not needed in this exact context, but this is the context of the message I left on slack

```
static const rosidl_type_hash_t MY_TYPE_HASH = {1, {0x00, ...}};
static const rosidl_typesupport_introspection_c__MessageMembers MY_TYPESUPPORT_INTROSPECTION = {
  ...,
  MY_TYPE_HASH,
  ...
}; 
// above OK for GCC, error C2099 for MSVC, following is OK for all platforms
#define MY_TYPE_HASH {1, {0x00, ...}}
```

The lvalue `MY_TYPE_HASH` can't be used in an aggregate initialization, as it's not a C89 const-expression. So, when I want to use this later in typesupport, that's when the problem arises. I figured having the initializer macro available doesn't hurt anybody as long as it's clear you can and should use the variable where possible

Reply:
Yeah, that's fine.  It just seemed like an unnecessary indirection, but it looks like because of MSVC, it is necessary indirection.  OK, no action to take here, thanks for the explanation.

Reply:
I added a brief comment to the template/generated code explaining why it is that way, for future readers.